pyfr/integrators/std/base.py

```diff
@@ -30,6 +25,9 @@ def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg):
         self._regidx = list(range(self.nregs))
         self._idxcurr = 0
 
+        # Pre-process solution if necessary
+        self.system.preproc(self.system.ele_scal_upts(self._idxcurr))
```

Comment:
Not sure if this will work?  As prepoc takes a register number whereas we're passing in a list of solutions?

Reply:
Updated.

Reply:
Also check the gradient calculation stuff, as it will also need the preproc logic if it does not have it already.